recipe/meta.yaml

```
@@ -26,21 +26,24 @@ outputs:
       build:
         - {{ compiler("c") }}
         - {{ compiler("cxx") }}
-        - {{ compiler("cuda") }}         # [cuda_compiler_version != "None"]
+        - {{ compiler("cuda") }}
         - {{ cdt("libnl") }}             # [cdt_name == "cos6"]
         - {{ cdt("libnl3") }}            # [cdt_name == "cos7"]
+        - cuda-nvml-dev ={{ cuda_compiler_version }}       # [cuda_compiler_version == "12.0"]
         - automake
         - autoconf
         - libtool
         - make
         - pkg-config
       host:
+        - cuda-nvml-dev ={{ cuda_compiler_version }}       # [cuda_compiler_version == "12.0"]
         - libnuma
         - rdma-core  # [linux64 and cuda_compiler_version != "10.2"]
       run:
       run_constrained:
```

As `cuda-version` is relatively lightweight and also [constrains `cudatoolkit`]( https://github.com/conda-forge/cuda-version-feedstock/blob/3ce1bae701079226f982b173667edc5d0a64c01f/recipe/meta.yaml#L17 ), think we can rely on it instead of `cudatoolkit` here. This will make it easier to use for CUDA 12+ versions as well

```suggestion
      run_constrained:
        - cuda-version {{ cuda_compiler_version }}|{{ cuda_compiler_version }}.*  # [cuda_compiler_version in ("10.2", "11.0", "11.1")]
        - cuda-version >=11.2,<12                                                 # [cuda_compiler_version == "11.2"]
        - cuda-version >=12,<13                                                   # [cuda_compiler_version == "12.0"]
```